tests/state/ViewerState/backdrop_slice.test.ts

```diff
@@ -0,0 +1,123 @@
+import { act, renderHook } from "@testing-library/react";
+import { describe, expect, it } from "vitest";
+
+import { Dataset } from "../../../src/colorizer";
+import { ANY_ERROR } from "../../test_utils";
+import { MOCK_DATASET_WITHOUT_BACKDROP } from "./constants";
+
+import { useViewerStateStore } from "../../../src/state/ViewerState";
+
+describe("useViewerStateStore: BackdropSlice", () => {
+  describe("setBackdropKey", () => {
+    it("does not allow backdrop slice to be set when provided Dataset has no backdrops", () => {
+      const { result } = renderHook(() => useViewerStateStore());
+
+      // Initialized as null
+      expect(result.current.backdropKey).toBeNull();
+
+      act(() => {
+        result.current.setBackdropKey(MOCK_DATASET_WITHOUT_BACKDROP, "test");
```

Comment:
why does this need to be wrapped in "act"? (I mean would you get the same result by calling it directly?)

Reply:
also, it seems to silently fail and just leave things as null.  is that basically the intent?  I'm not totally sure what is being tested here.

Reply:
Mentioned this above! For the react testing library, you have to wrap state changes in `act` to allow renders to occur.

Reply:
And yes, this currently ignores backdrop keys that aren't in the specified dataset. I guess I should throw an explicit error here instead, since that would indicate a frontend bug.